packages/flutter/test/rendering/debug_test.dart

```diff
@@ -124,4 +124,37 @@ void main() {
     expect(b.debugPaint, isNot(paints..rect(color: const Color(0x90909090))));
     debugPaintSizeEnabled = false;
   });
+
+  test('debugPaintPadding from render objects with inverted direction', () {
+    debugPaintSizeEnabled = true;
+    RenderSliver s;
+    final RenderViewport root = RenderViewport(
+      axisDirection: AxisDirection.up,
```

Comment:
In the PR description you said this also didn't work for left. Can you add a test for that direction as well? (And if we don't already have it, it may be worth it to add tests for the other two working directions as well).

Reply:
added left case, the normal direction is covered as part of previous test so we should be fine